rust/private/rustc.bzl

```diff
@@ -114,10 +116,20 @@ def collect_deps(deps, toolchain):
 
             # TODO: We could let the user choose how to link, instead of always preferring to link static libraries.
             libs = get_libs_for_static_executable(dep)
+            # TODO: This is changing the order of linker inputs, it shouldn't.
```

Comment:
The fix is to iterate through libs once, right?

Reply:
Correct. I didn't want to do that in this PR, but it should be fairly straightforward.